pkg/reconciler/pipelinerun/pipelinerun.go

```diff
@@ -407,12 +413,6 @@ func (c *Reconciler) reconcile(ctx context.Context, pr *v1beta1.PipelineRun) err
 		return controller.NewPermanentError(err)
 	}
 
-	if pipelineState.IsDone() && pr.IsDone() {
```

Comment:
`pr.IsDone()` only becomes true when the completion time is set, which happens down below in L461-466.
This is never invoked, as it is also proved by the fact that we don't get a duplicate event for this case.
The timeout is released in `ReconcileKind` when `isDone` is true.